### PR TITLE
Resource scm repo link

### LINKED ISSUE #56: Implement Resource "SCM Repo Link"
https://developers.scalingo.com/scm_repo_link

### PR DESCRIPTION
**Pull request by SCedricThomas**
Fix #56

### CODE REVIEW

File: scalingo/resource_scalingo_scm_repo_link.go [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+	}
+}
+
+func resourceScmRepoLinkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, _ := meta.(*scalingo.Client)
+
+	app, _ := d.Get("app").(string)
+	source, _ := d.Get("source").(string)
+	branch, _ := d.Get("branch").(string)
+	autoDeployEnabled, _ := d.Get("auto_deploy_enabled").(bool)
```
[the remark this answers is not in the document]
Comment (SCedricThomas) [reply scalingo/resource_scalingo_scm_repo_link.go:86]: I aligned the resource fields with the API fields. I also think that the new nomenclature is clearer

File: scalingo/resource_scalingo_scm_repo_link.go [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+	}
+
+	deployReviewAppsEnabled, _ := d.Get("deploy_review_apps_enabled").(bool)
+	deleteOnCloseEnabled, _ := d.Get("delete_on_close_enabled").(bool)
+	hoursBeforeDeleteOnClose, _ := d.Get("hours_before_delete_on_close").(int)
+	deleteStaleEnabled, _ := d.Get("delete_stale_enabled").(bool)
+	hoursBeforeDeleteStale, _ := d.Get("hours_before_delete_stale").(int)
+
+	if hoursBeforeDeleteOnClose < 0 || hoursBeforeDeleteStale < 0 {
+		return diag.Errorf("Hours must be an unsigned int")
```
Comment (EtienneM) [inline scalingo/resource_scalingo_scm_repo_link.go:101]:
```suggestion
		return diag.Errorf("hours must be an unsigned int")
```

**Review by EtienneM — APPROVED** [review]
🎉

File: scalingo/resource_scalingo_scm_repo_link.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Soulou) [reply scalingo/resource_scalingo_scm_repo_link.go:86]: I think I'm the one who pushed to have a "verb" and not just a noun as a field for booleans, to me it's a bit more explicit

**Pull request merged**